src/main/java/somethingrandom/dataaccess/google/tasks/TaskList.java

```diff
@@ -14,4 +14,8 @@ public class TaskList {
     public String getIdentifier() {
         return identifier;
     }
+
+    public APIProvider getProvider() {
+        return provider;
+    }
```

Comment:
This shouldn't exist! Your code to save items should go into a new `public void add(Item) throws DataAccessException` method that just uses the `provider` field.